module/exporter-file/src/Infrastructure/Handler/Export/DeleteExportCommandHandler.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\ExporterFile\Infrastructure\Handler\Export;
+
+use Ergonode\Channel\Domain\Query\ExportQueryInterface;
+use Ergonode\Exporter\Infrastructure\Service\ZipFileRemover;
+use Ergonode\Exporter\Domain\Command\Export\DeleteExportCommand;
+use Ergonode\ExporterFile\Domain\Entity\FileExportChannel;
+
+class DeleteExportCommandHandler
+{
+    private ZipFileRemover $zipFileRemover;
+
+
+    private ExportQueryInterface $exportQuery;
+
+    public function __construct(
+        ZipFileRemover $zipFileRemover,
+        ExportQueryInterface $exportQuery
+    ) {
+        $this->zipFileRemover = $zipFileRemover;
+        $this->exportQuery = $exportQuery;
+    }
+
+    public function __invoke(DeleteExportCommand $command): void
```

Comment:
You have 2 DeleteExportCommandHandler 
On like this one is dedicated for export-file module, second one is global without specification whit one is dedicated for.
That mean both of them will be executed.

We have 2 scenarios:

1)  first will be executed dedicated and second global that be fine
2) first will be executed global and second dedicated. In that case wee got error becous global delete export and dedicated try to load data from not exits export anymore.

i think both should be dedicated.

Reply:
Isn't the above handled with priorities? As long as the export is kept as a generic thing and usage of export repository suffice it's fine by me - and specific additional stuff(like file removal) can be handled by dedicated handlers